src/Control/IApplicative/Unified.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE ConstraintKinds       #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE KindSignatures        #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE StandaloneDeriving    #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+----------------------------------------------------------------------
+-- Now, can we define pure and <*> so they work on indexed OR
+-- non-indexed things?  With clever use of type families, perhaps?
+--
+-- Hmmm... really not sure whether this is possible.
+----------------------------------------------------------------------
+
```

Comment:
It was a (now abandoned) attempt at unifying `IApplicative` and `Applicative`.  It's not exported or listed in the `.cabal` file at all.